src/Objs/Product/ProductComponent.tsx

```diff
@@ -206,24 +212,60 @@ const CartActionButton = connect(function CartActionButton({
     )
   }
 
-  if (isInCart(product)) {
+  const quantity = quantityInCart(product)
+  if (quantity) {
     return (
-      <button
-        className="btn btn-sm btn-primary"
-        onClick={() => {
-          removeFromCart(product)
-          toast.info(cartRemovedMessage)
-        }}
-      >
-        <i className="bi bi-x-lg"></i>
-        {cartRemoveLabel}
-      </button>
+      <div className="row mt-0 mb-1 g-1">
+        <div className="col-auto">
+          <div className="input-group input-group-sm flex-nowrap">
+            <button
+              aria-label="-"
+              className="btn btn-primary"
+              disabled={quantity < 2}
+              onClick={() => subtractFromCart(product)}
+            >
+              <i className="bi bi-dash-lg px-0" />
+            </button>
+            <input
+              className="form-control text-center flex-grow-0 w-25"
+              defaultValue={quantity}
+              key={quantity}
+              min={1}
+              onBlur={async (e) => {
+                const quantity = Math.floor(Number(e.target.value))
+                if (quantity > 0) updateQuantityInCart(product, quantity)
+              }}
+              step={1}
+              type="number"
+            />
```

Comment:
Axe complains: `Ensure every form element has a label`. Please fix this:

![Screenshot 2025-03-04 at 12 39 17](https://github.com/user-attachments/assets/0734a8dc-4d70-4280-8c19-66e941c99e04)